### ISSUE TITLE
`--project-dir` with relative paths?

### ISSUE THREAD

**Issue by max-sixty** [fishtown-analytics/dbt#2541]
### Describe the bug
`--project-dir` doesn't use relative paths, and I got confused given the error message

### Steps To Reproduce
Run a project in a child path; say we have a DBT project in a path called `sql` under the current one:

```
$ dbt run --project-dir sql
Running with dbt=0.17.0
Encountered an error while reading the project:
  ERROR: Runtime Error
  no dbt_project.yml found at expected path sql/dbt_project.yml
Encountered an error:
Runtime Error
  Could not run dbt

$ dbt run --project-dir (pwd)/sql
Running with dbt=0.17.0
Found 4 models, 0 tests, 0 snapshots, 0 analyses, 147 macros, 0 operations, 0 seed files, 0 sources
```

I was doubting my sanity for a second with `no dbt_project.yml found at expected path sql/dbt_project.yml`

### Expected behavior
Use relative paths, or if this is deliberate, don't accept relative paths

**The output of `dbt --version`:**
```
installed version: 0.17.0
   latest version: 0.17.0

Up to date!

Plugins:
  - bigquery: 0.17.0
  - snowflake: 0.17.0
  - redshift: 0.17.0
  - postgres: 0.17.0
```

**The operating system you're using:**
MacOS

**The output of `python --version`:**
Python 3.7.7

**Comment by drewbanin** [comment]
hey @max-sixty - this tripped me up too! See here: https://github.com/fishtown-analytics/dbt/pull/2534#issuecomment-643272603

Let's definitely make relative paths work for `--project-dir` at some point in the future. This is a pretty tricky part of the codebase - dbt does lots of magic to support running `dbt run` from a subdir of a dbt project (eg. `models/`), so it might be a little involved to get this working well. I do think it's worth doing though!

**Comment by beckjake** [comment]
Fixed in #2534
